### PR TITLE
dashpay: 0.12.055 -> 0.12.1.3

### PR DESCRIPTION
**Pull request by ndowens**
###### Motivation for this change
Updated and added overridable options

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/altcoins/dashpay.nix [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+                  ++ optional with_gui qt4
+                  ;
 
-  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ];
+  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ]
+                    ++ optional with_gui "--with-gui=qt4"
+                    ++ optional (!with_gui) "--with-gui=no"
+                    ++ optional enable_upnp "--enable-upnp-default"
+                    ++ optional disable_wallet "--disable-wallet"
+                    ++ optional disable_daemon "--disable-wallet"
```
Comment (vcunat) [inline pkgs/applications/altcoins/dashpay.nix]: Copy&paste error?

File: pkgs/applications/altcoins/dashpay.nix [hunk 2]
```diff
@@ -1,25 +1,41 @@
-{ fetchzip, stdenv, pkgconfig, autoreconfHook
+{ fetchFromGitHub, stdenv, pkgconfig, autoreconfHook
 , openssl, db48, boost, zlib, miniupnpc
-, qt4, qrencode, glib, protobuf, yasm
-, utillinux }:
+, qrencode, glib, protobuf, yasm, libevent
+, utillinux
+, with_gui ? true, qt4
+, enable_upnp ? false
```
Comment (dezgeg) [inline pkgs/applications/altcoins/dashpay.nix]: Convention is to use `camelCase`, e.g. `withGui`.

Comment (ndowens) [inline pkgs/applications/altcoins/dashpay.nix]: camelCase ?

**Comment by ndowens** [comment]
I have lost access to this PR, had to open a new one #23761

**Pull request closed without merging**